radix-engine-common/src/types/type_identifier.rs

```diff
@@ -4,3 +4,8 @@ use sbor::LocalTypeIndex;
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Sbor)]
 pub struct TypeIdentifier(pub Hash, pub LocalTypeIndex);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Sbor)]
+pub enum TypeSubstitutionRef {
```

Comment:
`TypeRefSubstitution`?

Reply:
It's more like the "reference to the type I'd like to substitute" NOT "substitution of the type reference"